src/cargo/ops/cargo_test.rs

```diff
@@ -32,14 +32,15 @@ pub fn run_tests(manifest_path: &Path,
             Some(path) => path,
             None => exe.clone(),
         };
-        let cmd = try!(compile.process(exe, &compile.package)).args(test_args);
+        let cmd = try!(compile.process(CommandType::Target(exe.to_c_str()), &compile.package))
+                  .args(test_args);
         try!(options.compile_opts.shell.concise(|shell| {
             shell.status("Running", to_display.display().to_string())
         }));
         try!(options.compile_opts.shell.verbose(|shell| {
             shell.status("Running", cmd.to_string())
         }));
-        match cmd.exec() {
+        match ExecEngine::exec(&mut DefaultCommands, cmd) {
```

Comment:
The problem is that I'm hitting the same issues as above.
(see http://is.gd/wMU6hd)

The latest Rust implements trait Foo for types Foo, so I could use the `ExecEngine::exec` syntax with a `Box`.